subprojects/tooling-api/src/testFixtures/groovy/org/gradle/integtests/tooling/fixture/ToolingApiExecution.groovy

```diff
@@ -115,8 +110,11 @@ class ToolingApiExecution extends AbstractMultiTestRunner.Execution implements T
     }
 
     @Override
-    protected void before() {
-        def testClazz = testClassLoader.loadClass(ToolingApiSpecification.name)
+    protected void before(IMethodInvocation invocation) {
+        // TODO: how does this work? Does it work?
+        def testClazz = testClassLoader.loadClass(invocation.getSpec().getReflection().getName())
         testClazz.selectTargetDist(gradle)
+
+        ((ToolingApiSpecification)invocation.getInstance()).selectTargetDist(gradle)
```

Comment:
This does not work - as a result ToolingApi tests now always test against the current TAPI distribution from the current classpath.
I tried loading the whole `ToolingApi` class using the `testClassLoader`, which does load the right classes, and injecting it into the test class. Then the problem is that the loaded classes are not compatible with the ones used in the tests.
With the old approach using JUnit4, we passed the `testClassLoader` to JUnit and it used it to instantiate the test class for the test iteration. I was not able to find a way to do this with Spock2 - the test class is already instantiated and is being reused for subsequent iterations.